src/code_0807CC3C.c

```diff
@@ -0,0 +1,543 @@
+            }
+        } else {
+            r0 = 0;
+        }
+    }
+    return r0;
+}
+END_NONMATCH
+
+u32 sub_0807D0EC(u32 unk_1, const char* unk_2) {
+    u32 ret;
+
+    if (!sub_0807D1D8(unk_1, unk_2, 8)) {
+        ret = 0;
+    } else {
+        ret = sub_0807D128((Thing*)unk_2);
+    }
+    if (!ret && sub_0807D1D8(unk_1 + 8, unk_2, 8)) {
+        ret = sub_0807D128((Thing*)unk_2);
+    }
+    return ret;
+}
+
+u32 sub_0807D128(const Thing* thing) {
+    u32 ret;
+    switch (thing->unk_3) {
+        case 'MCZ3':
+            if (thing->unk_1 + thing->unk_2 == 0x10000) {
+                ret = 2;
+            } else {
+                ret = 0;
+            }
+            break;
+        case 'FleD':
+        case 'TINI':
+            ret = 0;
+            if ((thing->unk_1 & thing->unk_2) == 0xffff) {
+                ret = 1;
+            }
+            break;
+        default:
+            ret = 0;
+            break;
+    }
+    return ret;
+}
+
+u32 sub_0807D184(u32 param_1, const char* param_2) {
+    u32 uVar1;
+
+    uVar1 = sub_0807D20C(param_1, param_2, 8);
+    if (uVar1 == 0) {
+        uVar1 = sub_0807D20C(param_1 + 8, param_2, 8);
+    }
+    return uVar1;
+}
+
+u16 sub_0807D1A4(u16* unk_1, u32 unk_2) {
+    u32 uVar1;
+
+    uVar1 = 0;
+    while (unk_2 != 0) {
+        uVar1 = uVar1 + (*unk_1 ^ unk_2);
+        unk_1 = unk_1 + 1;
+        unk_2 = unk_2 - 2;
+    }
+    return uVar1;
+}
+
+struct_0807D1C4* sub_0807D1C4(u32 unk_1) {
+    return &gUnk_0811E4BC[unk_1];
+}
+
+// these three are basically the same and wrong by basically one instruction in the wrong place
+NONMATCH("asm/non_matching/code_0807CC3C/sub_0807D1D8.inc", u32 sub_0807D1D8(u32 unk_1, const char* unk_2, u32 unk_3)) {
+}
+END_NONMATCH
+
+NONMATCH("asm/non_matching/code_0807CC3C/sub_0807D20C.inc", u32 sub_0807D20C(u32 unk_1, const char* unk_2, u32 unk_3)) {
+    unk_3 >>= 3;
+    unk_1 >>= 3;
+    while (unk_3-- > 0) {
+        if (sub_080B18DC(unk_1, unk_2)) {
+            sub_080B18DC(unk_1, gUnk_0811E4B4);
+            return 0;
+        }
+        unk_1++;
+        unk_2 += 8;
+    }
+    return 1;
+}
+END_NONMATCH
+
+NONMATCH("asm/non_matching/code_0807CC3C/sub_0807D24C.inc", u32 sub_0807D24C(u32 unk_1, const char* unk_2, u32 unk_3)) {
+    unk_3 >>= 3;
+    unk_1 >>= 3;
+    while (unk_3-- > 0) {
+        if (sub_080B180C(unk_1, unk_2))
+            return 0;
+        unk_1++;
+        unk_2 += 8;
+    }
+    return 1;
+}
+END_NONMATCH
+
+// these three functions use gRoomControls, maybe once that is understood better, these can be decompiled easier
+NONMATCH("asm/non_matching/code_0807CC3C/sub_0807D280.inc", void sub_0807D280(u32 unk_1, u32 unk_2)) {
+}
+END_NONMATCH
+
+NONMATCH("asm/non_matching/code_0807CC3C/sub_0807D46C.inc", void sub_0807D46C(u32 unk_1, u32 unk_2)) {
+}
+END_NONMATCH
+
+NONMATCH("asm/non_matching/code_0807CC3C/sub_0807D6D8.inc", void sub_0807D6D8(u32 unk_1, u32 unk_2)) {
+}
+END_NONMATCH
+
+void sub_0807DA70(void) {
```

Comment:
All of this script stuff can probably be moved into the executeScriptCommandSet.c file (might need a name change to just script.c) since it is adjacent and was probably in that TU originally.

Reply:
good idea, any suggestions on the new filename

Reply:
the first few functions probably should be moved to save.c too

Reply:
In that case perhaps you can just split them between save.c and script.c and delete the old one

Reply:
there are some functions inbetween that I'm not sure in which one thy belong, if any

Reply:
looked at them in Ghidra and they seem like they might be unused -- there's a pointer table which references those particular three functions but it has no direct references (and I couldn't trigger a breakpoint when playing for a couple mins).

I would just put them in script.c for the time being.

Reply:
I don't think just moving them somewhere without knowing what they are about is a good idea, and all I can tell so far is, that they do a bunch of stuff with gRoomControls

Reply:
If you would like to leave them in the file then that should be fine.